### ISSUE TITLE
Examples?

### ISSUE THREAD

**Issue by kkukshtel** [NoelFB/Foster#9]
Would love to see some really simple examples! Doesn't have to be anything robust, but would love to see a little more than how to render a rectangle. Maybe an overview of main API stuffs?

**Comment by NoelFB** [comment]
Perhaps creating a Wiki for that stuff on github makes sense? For API documentation and simple examples.

**Comment by kkukshtel** [comment]
Yeah definitely! I look to something like [DOME ](https://domeengine.com/)as a possible template. It's documentation is definitely not robust, but there's enough documentation there of most of the core methods/etc. that you can get started.

**Comment by NoelFB** [comment]
Cool, I'll take a look!

I could also start by generating API docs. Most of the C# methods are commented so if there's a tool that generates docs from that I can probably easily set up an API page to begin with.

**Comment by kkukshtel** [comment]
Yeah I was definitely going to suggest that. I know Doxygen is the one people typically recommend but I find it hard to read sometimes. Not sure if there is something C# specific and nice.

**Comment by NoelFB** [comment]
Yeah Doxygen sort of feels like overkill to me. Ideally I'd like something that can live on the repo (either as a Wiki or as Github Pages in a Docs) folder).

Maybe instead of a full API it would make sense to just write out explanations of the core modules and general info on what each one does, and how to use them. The Github Wiki seems pretty idea for that kind of thing.

**Comment by mourelask** [comment]
Maybe [docfx](https://github.com/dotnet/docfx) ? It generates static html websites.

**Comment by kkukshtel** [comment]
docfx definitely looks cool. I think it's kind of up to @NoelFB though on how much work you want to put into it right now as the person who obviously knows the most :) Writing up a wiki is easier to start, especially if the APIs are changing a lot, the autogen stuff can quickly go out of date.

**Comment by NoelFB** [comment]
I've started working on a simple Wiki to handle the basics. I think at some point it'd be good to have an API that's generated from the C# documentation but it probably makes sense for the API to solidify a bit more first :)

**Comment by kkukshtel** [comment]
This looks really nice, thanks so much for writing it up!

I think one thing that would be nice to add would be a sort of "philosophy" of how Foster wants to make a game. Like in reading it I'm not sure if I should do the whole game in a single module, how the modules talk to each other (or if they should), etc.

At least something like that in lieu of actual simple demo game.

It also reminds me of an idea I had a bit ago to try to make [Derek Yu's Space Shooter Gamemaker Tutorial](https://forums.tigsource.com/index.php?topic=3251.0) become the Lorem Ipsum of game frameworks. Maybe I'll clone this down over the weekend and try to make it happen.

**Comment by NoelFB** [comment]
Yeah that's a good idea. I should write out a simple game example.

I'm making a separate "game engine" project that sits on top of this, for handling Assets, Collisions, Scene/Entity/Component system, etc. But it would still be nice to write out an explanation for making stuff without that.

**Comment by Colbydude** [comment]
Hello~

Came across this issue a while back and saw some of Noel's stream making [tiny_link](https://github.com/NoelFB/tiny_link) two weeks or so ago. Been itchin' to do some gamedev and thought it'd be fun porting the game from blah to Foster for both practice and having a simple game to showcase Foster. You can checkout the repo here: https://github.com/Colbydude/FosterPlatformer

Feel free to take it and modify however. If anyone has time, I'd super appreciate feedback on the port, as my C++/C# chops are a bit rusty (I come from primarily a webdev space). Happy to answer any questions as well!

I've also forked Foster to add a couple of Time utilities (just `PauseFor()` and `OnTime()` and would be happy to open a PR to get those added, assuming I've implemented correctly. :)

Cheers!
